tests/nix/default.nix

```diff
@@ -0,0 +1,131 @@
+{
+  self,
+  config,
+  lib,
+  inputs,
+  ...
+}:
+let
+  inherit (lib)
+    mkOption
+    mapAttrs'
+    mapAttrsToList
+    flatten
+    ;
+  inherit (lib.types)
+    submodule
+    lines
+    attrsOf
+    deferredModule
+    lazyAttrsOf
+    ;
+  cfg = config.wire.testing;
+in
+{
+  imports = [ ./suite/test_basic_deploy ];
+  options.wire.testing = mkOption {
+    type = attrsOf (
+      submodule (
+        { name, ... }:
+        {
+          options = {
+            nodes = mkOption {
+              type = lazyAttrsOf deferredModule;
+            };
+            testScript = mkOption {
+              type = lines;
+              default = ''
+
+              '';
```

Comment:
You should probably include basic sane tests, such as starting all nodes, sanity testing ssh connection and waiting for the basic services.

Reply:
that's.... why it is still a draft PR

Reply:
i read Nitpicks are welcome as an invitation to Have at It lol

Reply:
> i read Nitpicks are welcome as an invitation to Have at It lol

no it's fine i was in a bad mood, that's my fault.

Reply:
> We should probably include basic sane tests, such as starting all nodes, sanity testing ssh connection and waiting for the basic services.

can you elaborate more on this?

- [x] starting all nodes by default
- [ ] sanity testing ssh connection (I basically disabled the strict host key checking so does it matter now?)
- [x] waiting for basic services, I suppose you mean sshd.service. that is done